pkg/cmd/server/kubernetes/node/node_config.go

```diff
@@ -374,9 +375,18 @@ func BuildKubernetesNodeConfig(options configapi.NodeConfig, enableProxy, enable
 	return config, nil
 }
 
-func buildKubeProxyConfig(options configapi.NodeConfig) (*proxyoptions.ProxyServerConfig, error) {
+func buildKubeProxyConfig(options configapi.NodeConfig) (*componentconfig.KubeProxyConfiguration, error) {
+	proxyOptions, err := kubeproxyoptions.NewOptions()
+	if err != nil {
+		return nil, err
+	}
+	// Resolve cmd flags to add any user overrides
+	if err := cmdflags.Resolve(options.ProxyArguments, proxyOptions.AddFlags); len(err) > 0 {
```

Comment:
Why did this get moved to the start of the function? The goal is that options.ProxyArguments is supposed to override the defaults that get set here. In particular, you're supposed to be able to add:

    proxyArguments:
      proxy-mode: ["userspace"]

to node-config.yaml to override the default value of `proxyconfig.Mode` that gets set below.